src/main/java/com/box/sdk/retention/RetentionPolicy.java

```diff
@@ -146,18 +204,33 @@ public void setRetentionLength(String retentionLength) {
 
 		@Override
 		public boolean equals(Object o) {
-			if (this == o) return true;
-			if (o == null || getClass() != o.getClass()) return false;
+			if (this == o) {
```

Comment:
It's good practice to implement hashCode when implementing equals in Java ([reference](http://stackoverflow.com/questions/17027777/relationship-between-hashcode-and-equals-method-in-java)). We can either generate with intelliJ or use apache-commons to do ReflectionEquals and ReflectionHashCode. I'll look into this.